### PR TITLE
Fix, Test requestUtil prefs/device; add helpers

### PR DESCRIPTION
**Pull request by ayumi**
Auditors: @diracdeltas

### CODE REVIEW

File: client/sync.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ const maybeSetDeviceId = () => {
   if (!requester.s3) {
     throw new Error('cannot set device ID because AWS SDK is not initialized.')
   }
-  return requester.list(proto.categories.PREFERENCES)
+  return requester.list(proto.categories.wS)
```
Comment (diracdeltas) [inline client/sync.js]: wat is `wS`?

File: client/sync.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ const logSync = (message, logLevel = DEBUG) => {
  * @returns {Object}
  */
 const decrypt = (ciphertext) => {
-  const d = clientSerializer.byteArrayToSecretBoxRecord(ciphertext)
+  const d = clientSerializer.byteArrayToSecretboxRecord(ciphertext)
```
Comment (diracdeltas) [inline client/sync.js]: oops good catch

File: test/client/requestUtil.js [hunk 3]
```diff
@@ -23,14 +26,71 @@ test('client RequestUtil', (t) => {
...
+    const serializer = requestUtil.serializer
+    const decrypt = testHelper.Decrypt(serializer, keys.secretboxKey)
+    const encrypt = clientTestHelper.Encrypt(serializer, keys.secretboxKey)
+
     t.plan(1)
-    t.assert(requestUtil)
+    t.test('#put preference: device', (t) => {
+      t.plan(2)
+      const deviceId = new Uint8Array([0])
+      const name = 'alpha pyramid'
```
Comment (diracdeltas) [inline test/client/requestUtil.js]: :pray:

File: client/requestUtil.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 'use strict'
 
 const awsSdk = require('aws-sdk')
-const crc32 = require('buffer-crc32')
+const crc = require('crc')
+const s3Helper = require('../lib/s3Helper')
 
 // Max size of a record in bytes. Limited by s3 key size.
 const MAX_RECORD_SIZE = 900
```
Comment (diracdeltas) [inline client/requestUtil.js]: should this be smaller now that the encoding is more chars?

File: client/sync.js (hunk 1 above)
Comment (ayumi) [inline client/sync.js]: oops typo

**Comment by diracdeltas** [comment]
fixes to be done in follow ups

**Pull request merged**